.github/workflows/sync_docs.yml

```diff
@@ -0,0 +1,61 @@
+name: Update dev docs
+
+on:
+  push:
+    branches:
+      - main
+      - 13.0.x
+    paths:
+      - documentation
+
+env:
+  target: git@github.com:infinispan/infinispan.github.io.git
+  target_branch: master
+
+jobs:
+  build-dev:
+    if: github.ref == 'refs/heads/main'
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up JDK 11
+      uses: actions/setup-java@v2
+      with:
+        java-version: '11'
+        distribution: 'temurin'
+        cache: maven
+    - name: Build docs with Maven
+      run: mvn install -Pdistribution -pl documentation -am -s maven-settings.xml
+    - name: Copy docs to dev
+      run: |
+        git clone -b $target_branch $target
+        cp -r infinispan/documentation/target/generated/1*/html/* infinispan.github.io/docs/dev/
+        cd infinispan.github.io
+        git config --global user.email "dnaro@redhat.com"
+        git config --global user.name "dnaro"
+        git add . --all
+        git commit -m "Synchronized docs to dev"
+        git push origin master
+  build-stable:
+    if: github.ref == 'refs/heads/13.0.x'
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up JDK 11
+      uses: actions/setup-java@v2
+      with:
+        java-version: '11'
+        distribution: 'temurin'
+        cache: maven
+    - name: Build docs with Maven
+      run: mvn install -Pdistribution -pl documentation -am -s maven-settings.xml
+    - name: Copy docs to stable
+      run: |
+        git clone -b $target_branch $target
+        cp -r infinispan/documentation/target/generated/1*/html/* infinispan.github.io/docs/stable/
+        cd infinispan.github.io
+        git config --global user.email "dnaro@redhat.com"
+        git config --global user.name "dnaro"
+        git add . --all
+        git commit -m "Synchronized docs to stable"
+        git push origin master
```

Comment:
I think we can reduce repetition further by doing something like:

```yaml
name: Update dev docs

on:
  push:
    branches:
      - main
      - 13.0.x
    paths:
      - documentation

env:
  target: git@github.com:infinispan/infinispan.github.io.git
  target_branch: master

jobs:
  upload:
    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v2
    
    - name: Set up JDK 11
      uses: actions/setup-java@v2
      with:
        java-version: '11'
        distribution: 'temurin'
        cache: maven
    
    - name: Build docs with Maven
      run: mvn install -Pdistribution -pl documentation -am -s maven-settings.xml

    - name: Clone infinispan.github.io
      run: |
        git clone -b $target_branch $target

    - if: github.ref == 'refs/heads/main'
      name: Copy docs to dev
      run: |
        cp -r infinispan/documentation/target/generated/1*/html/* infinispan.github.io/docs/dev/

    - if: github.ref == 'refs/heads/13.0.x'
      name: Copy docs to stable
      run: |
        cp -r documentation/target/generated/1*/html/* infinispan.github.io/docs/stable/
        
    - name: Commit and push changes
      run: |
        cd infinispan.github.io
        git config --global user.email "dnaro@redhat.com"
        git config --global user.name "dnaro"
        git add . --all
        git commit -m "Synchronized docs from infinispan ${{ github.ref }}"
        git push origin master
```

Reply:
Oh that is top class. Updated!!